packages/itwinui-react/src/core/FileUpload/FileUpload.tsx

```diff
@@ -69,10 +62,7 @@ export const FileUpload = React.forwardRef((props, ref) => {
     e.stopPropagation();
 
     // only set inactive if secondary target is outside the component
-    if (
-      isDragActive &&
-      !fileUploadRef.current?.contains(e.relatedTarget as Node)
-    ) {
+    if (isDragActive && !e.currentTarget?.contains(e.relatedTarget as Node)) {
```

Comment:
* To confirm, what is the `relatedTarget` in dragLeave events?
* What is the difference between `e.target` and `e.currentTarget`?

Reply:
> * To confirm, what is the `relatedTarget` in dragLeave events?
> * What is the difference between `e.target` and `e.currentTarget`?

The only thing that changed in this PR is `fileUploadRef.current` → `e.currentTarget`. The [`currentTarget` property](https://developer.mozilla.org/en-US/docs/Web/API/Event/currentTarget) is always the element on which the event handler is attached, so it avoids the need to use a ref.

`target` is usually not relevent unless using [event delegation](https://javascript.info/event-delegation) (which we don't). And for `relatedTarget`, you need to look at some [very specific docs](https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/relatedTarget).

> Also, do you recommend any particular docs to better understand the drag-drop API?

MDN has a good AI-free writeup: https://developer.mozilla.org/en-US/docs/Web/API/HTML_Drag_and_Drop_API/File_drag_and_drop